src/enki_checks.py

```diff
@@ -1,8 +1,29 @@
 import re
+import itertools
 
 from enki_msg import Report
 from enki_regex import Regexes, Tags
 
+# standalone test to run on code blocks
+def sudo_check(
+    stripped_file: str,
+    report: Report,
+    file_path: str) -> None:
+    """Checks if sudo access is mentioned in documentation"""
+    code_block_dots = Regexes.CODE_BLOCK_DOTS.findall(stripped_file)
+    code_block_dashes = Regexes.CODE_BLOCK_DASHES.findall(stripped_file)
+    code_block_two_dashes = Regexes.CODE_BLOCK_TWO_DASHES.findall(stripped_file)
+
+    code_block = code_block_dots + code_block_dashes + code_block_two_dashes
+
+    # regex for root access  "# ", "~]# ", "*# ", " # ", "[root@", "[command]`# ", "#{nbsp}", "pass:quotes[#"
+    for item in ["$ sudo ", "$ *sudo ", "su -", "*su -"]:
```

Comment:
To catch more cases than these four, I've constructed this regular expression that takes into account text like the directory name in front of the prompt, the dollar sign as well as the hash sign, or no prompt at all, etc.:

```
^((\S+\])?[$#]\s+)?sudo|su\b
```

See interactively:

https://regex101.com/r/aabZYK/5

In my experience, these are realistic cases that we could potentially catch in the RHEL repo.

However, it doesn't take into account these cases yet:

```
[command]`#
```

```
pass:quotes[#
```

Reply:
I agree with you. I did that initially try to catch all the cases including # (the output was over 1k files) but Tomas wants to only catch sudo and leave the # and root be for now. I'll add you regex to the commented out portion for when we're ready to address the problem

Reply:
> In my experience, these are realistic cases that we could potentially catch in the RHEL repo.

I agree with you. I did that initially try to catch all the cases including # (the output was over 1k files) but Tomas wants to only catch sudo and leave the # and root be for now. I'll add you regex to the commented out portion for when we're ready to address the problem

Reply:
> ^((\S+\])?[$#]\s+)?sudo|su\b

Talked to Tomas, let's put ur regex in